### ISSUE TITLE
string_concat fails when concatenating working_directory

### ISSUE THREAD

**Issue by Grisgram** [YoYoGames/GameMaker-Bugs#6480]
### Description
Hey there,
maybe `working_directory` is internally a zero-terminated string and this causes `string_concat` to stop processing arguments, but the attached project shows the bug.

```

// LOOK IN THE LOG TO SEE THE EFFECT!

#macro SOME_TEXT		"some_text"
#macro WORLD			"world"

// This works
var s = string_concat("Hello ", WORLD, " here is ", SOME_TEXT);
show_debug_message(s);

// This not!
// (Either it's an effect for ALL "green" variables or just for the working_directory
// However, all my file access functions are broken now...
// Please hotfix!

s = string_concat(working_directory, SOME_TEXT);
// You see only the working directory in the log and NOT SOME TEXT!
// string_concat stops or breaks after adding the working_directory... maybe it is zero-terminated??
show_debug_message("HERE IS THE FAILED TEXT: [" + s + "]");
```

log output:
```
Hello world here is some_text
HERE IS THE FAILED TEXT: [C:\Users\****\AppData\Local\GameMakerStudio2\GMS2TEMP\string_concat_bug_48F5C944_VM\]
```
The second line SHOULD also contain "some_text" but string_concat aborted after the `working_directory`
Please fix - all my file access libs are broken now 

## Update
After investigating a bit further, the problem is bigger.
`string_concat` fails in many situations, even in simple string concats like this:
```
string_concat(LG_ROOT_FOLDER, __LG_LOCALE_BASE_NAME, localeName, DATA_FILE_EXTENSION)
```
where the macros used here are simple strings, but one of them configuration-dependant:
```
#macro LG_ROOT_FOLDER					"locale/"
#macro __LG_LOCALE_BASE_NAME	"locale_"
`localeName` contains "en" (function argument)
#macro DATA_FILE_EXTENSION				".json"
#macro release:DATA_FILE_EXTENSION		".jx"
```
The call above conatenates everything except the DATA_FILE_EXTENSION. It stops after locale_name.

thanks for reading!
Cheers, Gris

PS: I have uninstalled and rolled back to 2404 until this is fixed, I tried to work around but i have over 100 places where string_concat is used and this is impossible. Need to stick with 2404.

### Steps To Reproduce
* string_concat anything (macros, variables, string literals)... all good
* string_concat anything containing the green variable `working_directory` and the concat stops after the working directory. no other arguments are processed

I only tested this on windows currently, but please check all platforms for this, it might affect others too!

### Which version of GameMaker are you reporting this issue for?
IDE v2024.6.0.157 Runtime v2024.6.0.205

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

### Which platform(s) are you seeing the problem on?
Windows

[6820f0d5-0efc-4b83-936c-583275a1fd6e](https://api.gamemaker.io/api/github/downloads/6820f0d5-0efc-4b83-936c-583275a1fd6e)

**Comment by rwkay** [comment]
Bug fixed in 2024.8 and cherry picked to 2024.6.1

**Comment by cameron-home** [comment]
Verified fixed in IDE v2024.800.0.587  Runtime v2024.800.0.611

**Comment by YYDan** [comment]
Both functions now work fine in  IDE v2024.6.1.159  Runtime v2024.6.1.207:

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/15669932/7625d60b-3011-45e8-b054-42bf1b112add)

[string_concat bug-Dan.zip](https://github.com/user-attachments/files/16111165/string_concat.bug-Dan.zip)
